controller/controller.go

```diff
@@ -165,7 +164,36 @@ func (c *Impl) EnqueueLabelOf(namespaceLabel, nameLabel string) func(obj interfa
 				return
 			}
 
-			controllerKey = fmt.Sprintf("%s/%s", controllerNamespace, controllerKey)
+			c.EnqueueKey(fmt.Sprintf("%s/%s", controllerNamespace, controllerKey))
+			return
+		}
+
+		// Pass through namespace of the object itself if no namespace label specified.
+		// This is for the scenario that object and the parent resource are of same namespace,
+		// e.g. to enqueue the revision of an endpoint.
+		c.EnqueueKey(fmt.Sprintf("%s/%s", object.GetNamespace(), controllerKey))
+	}
+}
```

Comment:
Could you leave a `// DEPRECATED ... ` alias with the name `EnqueueLabelOf` until we migrate the downstream repos?  Otherwise this LGTM (though the method name is a mouthful).

Reply:
That would be friendly to backward compatibility, while the shortcoming is that we'll need to cleanup it later.

To avoid conflict, I was intending to update the upstream usage once this is merged.
But let's leave the deprecated one since you feel that better too.

Reply:
done